packages/terminal-bootstrap/lib/app.js

```diff
@@ -0,0 +1,57 @@
+//
+// SPDX-FileCopyrightText: 2022 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+'use strict'
+
+const EventEmitter = require('events')
+const pTimeout = require('p-timeout')
+
+class App extends EventEmitter {
+  constructor (informers, { timeout = 30_000 } = {}) {
+    super()
+    this.ac = new AbortController()
+    this.informers = informers
+    this.timeout = timeout
+  }
+
+  run () {
+    const untilHasSyncedList = []
+    for (const informer of Object.values(this.informers)) {
+      informer.run(this.ac.signal)
+      untilHasSyncedList.push(informer.store.untilHasSynced)
+    }
+    const fulfilled = () => this.emit('ready')
+    const rejected = err => {
+      this.shutdown()
+      this.emit('error', err)
+    }
+    pTimeout(Promise.all(untilHasSyncedList), this.timeout).then(fulfilled, rejected)
```

Comment:
as this is tied to lightship signalReady / shutdown, it would be nice to see from the logs which informer was synced / which one didn't or e.g. when the app is shotdown. Currently it just more or less silently shuts down, or not?

Reply:
✔️